### ISSUE TITLE
potentially insufficient UIDs or GIDs available in user namespace

### ISSUE THREAD

**Issue by mydeveloperplanet** [containers/podman-desktop#3444]
### Bug description

When building an image via Podman Desktop, I encounter the following error:
`Uploading the build context from /home/.../mypodmanplanet...Can take a while...
Error:(HTTP code 500) server error - potentially insufficient UIDs or GIDs available in user namespace (requested 262143:262143 for /var/tmp/libpod_builder2108531042/bError:Error: (HTTP code 500) server error - potentially insufficient UIDs or GIDs available in user namespace (requested 262143:262143 for /var/tmp/libpod_builder2108531042/build/.git): Check /etc/subuid and /etc/subgid: lchown /var/tmp/libpod_builder2108531042/build/.git: invalid argument `

I can build the image without any problem via CLI: `podman build . --tag mydeveloperplanet/mypodmanplanet:0.0.1-SNAPSHOT -f podman-desktop/1-Dockerfile`

I have extended the UIDs and GIDs for my user. It seems that Podman Desktop is running as a specific other user?

### Operating system

Ubuntu 22.04

### Installation Method

Installer from website/GitHub releases

### Version

1.2.1
I am using Podman 3.4.4

### Steps to reproduce

clone the repo: https://github.com/mydeveloperplanet/mypodmanplanet

build the jar-file: `mvn clean verify`

Use Podman Desktop to build the image:
Dockerfile: podman-desktop/1-Dockerfile
Build context: should be the root of the repository
Image Name: mydeveloperplanet/mypodmanplanet:0.0.1-SNAPSHOT
click Build

I am able to build the image without any problem via CLI:
`podman build . --tag mydeveloperplanet/mypodmanplanet:0.0.1-SNAPSHOT -f podman-desktop/1-Dockerfile`

### Relevant log output

```shell
see above
```

**Comment by benoitf** [comment]
hello @mydeveloperplanet 

Could you check if you're using a rootful podman or rootless with the CLI

On Linux Podman Desktop is always using the rootless connection

**Comment by mydeveloperplanet** [comment]
The rootless version, checked it with `podman system info`. I also see that the gidmap and sidmap are ok, I updated them when I was experimenting with Podman CLI.
```
idMappings:
    gidmap:
    - container_id: 0
      host_id: 1688000513
      size: 1
    - container_id: 1
      host_id: 200000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1688001497
      size: 1
    - container_id: 1
      host_id: 200000
      size: 65536
```

**Comment by Nezteb** [comment]
I have this same problem on macOS Ventura 13.6 (M2 Pro) using Podman Desktop 1.4.0 (using Podman 4.6.2).

```
❯ docker run -p 7687:7687 -p 7444:7444 -p 3000:3000 --name memgraph memgraph/memgraph-platform
Resolving "memgraph/memgraph-platform" using unqualified-search registries (/etc/containers/registries.conf.d/999-podman-machine.conf)
Trying to pull docker.io/memgraph/memgraph-platform:latest...
Getting image source signatures
Copying blob sha256:62707a86e6eb72f14f0dbe58a7da728b7586ab1b135c6071b0e6ba2b139d6a82
Copying blob sha256:31f5dc1f52c865588c43d8ec718f14d430e149b28f0b28232da825da7ae28f76
Copying blob sha256:28f6f99c2a0b36ff8c315f193e37e8e8c2151cbed4628fc4e300899082b461c2
Copying blob sha256:a1f53eb6dcf4da9e8ec7de93fd098766f4614b8db0deb8e81939c8e51c407581
Copying blob sha256:567d5e09580aa2ecaeaf0ca4ffeb76298fa7495a9095d32b67126c3f716f543b
Copying blob sha256:2f37bce1922c6c843ac9899e2cb091c4521fdbf21e8fae401712a623a0edabcf
Copying blob sha256:fb84e46c136cc20bb62d5f6d359c573a1176fefdebf8be16275be4b4f958f786
Copying blob sha256:000c713cf09cadde354be9779e9df7da25e4637006cd45c4e767a07a70f58904
Copying blob sha256:1bbbc71e3742bee36aeb61b30f92dd0f85dbc726cf2d5f6e258c2b1890f9d1b9
Copying blob sha256:bc2f64460a583010084cf68e8dd19b6a4aec5188757e5c32a72d197d9941e7cc
Copying blob sha256:590f0f85be57af07ec1ade2e1dc1b3fd09536feabe9c0f42add4fa82e4000789
Copying blob sha256:91d4800558c2d7585605f0ecb4a1f063aadf6c293fa38018e372d46cf57b01cc
Copying blob sha256:9181844769c72b4f456ac1637bb28ede1dec29465c3139d9f7f7cc8ac247b0c6
Copying blob sha256:3cdfa2e4c92b4391fdf933369cf3b97f728ed5133c250d8922b77c0ae4665287
Copying blob sha256:6a6d3e75d4f985d02c2e39eea3d2b661e64b50e0f9e20a1f125525e21f633ce4
Copying blob sha256:f86d2ddaf18a13d4229a86cc3f5ee0b119e781117d0418fb0555bc80d39ed80f
Copying blob sha256:907f4551a086c1da2e8e099b8f70df9e1576bffc93b3870d80da32886ee97691
Copying blob sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1
Error: copying system image from manifest list: writing blob: adding layer with blob "sha256:2f37bce1922c6c843ac9899e2cb091c4521fdbf21e8fae401712a623a0edabcf": processing tar file(potentially insufficient UIDs or GIDs available in user namespace (requested 718322462:454177323 for /lab/node_modules/buffer-equal-constant-time/.npmignore): Check /etc/subuid and /etc/subgid if configured locally and run "podman system migrate": lchown /lab/node_modules/buffer-equal-constant-time/.npmignore: invalid argument): exit status 1

❯ podman system migrate
Error: unrecognized command `podman system migrate`
Try 'podman system --help' for more information
```

EDIT: Disregard, I had to run `podman machine set --rootful` for this particular image to pull/build apparently? The error message didn't indicate anything related to root/permissions. 😅

**Comment by acancell-redhat** [comment]
Asking to reopenthis issue @benoitf :

ENV:

- podman desktop: 1.41.1
- podman version
~~~
Client:       Podman Engine
Version:      5.2.3
API Version:  5.2.3
Go Version:   go1.22.7
Built:        Tue Sep 24 02:00:00 2024
OS/Arch:      linux/amd64
~~~

ISSUE:

~~~
$ curl --unix-socket /run/user/4204350/app/io.podman_desktop.PodmanDesktop/podman-remote-ibmpower-machine.sock -X POST -H "Content-Type:application/tar" --data-binary '@build-context.tar' "http://v1.47.0/build"
{"cause":"invalid argument","message":"potentially insufficient UIDs or GIDs available in user namespace (requested 4204350:4204350 for /var/tmp/libpod_builder108973413/build/Containerfile): Check /etc/subuid and /etc/subgid if configured locally and run \"podman system migrate\": lchown /var/tmp/libpod_builder108973413/build/Containerfile: invalid argument","response":500}
~~~

ANALYSIS:

While instead:
~~~
$ tar --owner=0 --group=0 --no-same-owner --no-same-permissions -cvf build-context.tar Containerfile  index.html  
Containerfile
index.html
~~~
and then:
~~~
$ curl --unix-socket /run/user/4204350/app/io.podman_desktop.PodmanDesktop/podman-remote-ibmpower-machine.sock -X POST -H "Content-Type:application/tar" --data-binary '@build-context.tar' "http://v1.47.0/build?dockerfile=Containerfile "
...
{"stream":"Successfully built 0f9d038b6a35\n"}
~~~

INTERNAL DISCUSSIONS:
- https://redhat-internal.slack.com/archives/C032QG6EQE7/p1733709482791439
-  https://redhat-internal.slack.com/archives/CBBJY9GSX/p1733739387511199

**Comment by slemeur** [comment]
thanks @benoitf
